### PR TITLE
Fix CI Tests and Artifact Cache Issues

### PR DESCRIPTION
**Pull request by fchorney**
Use CLI script to automatically replace the 1.3 test with a 1.5 test (if necessary), as well as use new cache steps to cache the julia build artifacts

### CODE REVIEW

**Comment by nicoleepp** [comment]
Myabe @omus has an opinion but if tests fail on 1.0.x we should prob bump the Project.toml julia version that we say this package works on?

**Comment by omus** [comment]
The package was working on 1.0. I'll check that out tomorrow

**Comment by omus** [comment]
The 1.0 failure is a flaw in the YAML.jl package. I'm addressing it

**Comment by omus** [comment]
Here's the fix: https://github.com/JuliaData/YAML.jl/pull/109

**Comment by omus** [comment]
YAML 0.4.6 is released so you should be able to run the tests on Julia 1.0 again

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -15,46 +15,44 @@ jobs:
...
         include:
-          # Add a 1.0 job just to make sure we still support it
-          - os: ubuntu-latest
-            version: 1.0.5
-            arch: x64
-          # Add a 1.3 job because that's what Invenia actually uses
+          # Add a 1.5 job because that's what Invenia actually uses
           - os: ubuntu-latest
-            version: 1.3
+            version: 1.5
```
Comment (omus) [inline .github/workflows/CI.yml]:
```suggestion
            version: "1.5"
```

**Review by omus — APPROVED** [review]

**Pull request merged**